Qt/Debugger: Memory View font based sizing, row coloring

TryTwo opened the pull request:
Allows more rows to fit on screen.
Added alternating row colors. I can remove it if disliked.
Added a mark to always indicate the middle (address focused) row. I didn't want to mark the whole row, so the BP box is just darker grey.

BhaaLseN commented on Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp:
```diff
@@ -32,8 +32,14 @@ MemoryViewWidget::MemoryViewWidget(QWidget* parent) : QTableWidget(parent)
+
+  // Row height as function of text size. Less height than default.
+  const int fonth = fm.height() + 3;
+  verticalHeader()->setMaximumSectionSize(fonth);
```
Why is this `fm.height() + 3` when #7697 only uses `fm.height() + 1` for `verticalHeader()->setMaximumSectionSize(...)`?

TryTwo replied:
Memory View being a series of numbers is too cramped and hard to decipher with a very tight row height. Code view looks okay to me.

TryTwo commented:
![memoryview](https://user-images.githubusercontent.com/10532806/51074049-261da480-1637-11e9-9403-72a29c374cb2.jpg)

TryTwo commented:
Oops, forgot to change the Update loop using a constant row height. Also fonth -> rowh.

TryTwo commented:
And forgot to set variable BP column/icon size. Should be all done now.

BhaaLseN commented on Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp:
```diff
@@ -185,19 +194,23 @@ void MemoryViewWidget::Update()
-                       Resources::GetScaledThemeIcon("debugger_breakpoint").pixmap(QSize(24, 24)));
+      bp_item->setData(
+          Qt::DecorationRole,
+          Resources::GetScaledThemeIcon("debugger_breakpoint").pixmap(QSize(rowh, rowh)));
```
Is it a good idea to scale this icon?

lioncash commented on Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp:
```diff
@@ -32,8 +32,14 @@ MemoryViewWidget::MemoryViewWidget(QWidget* parent) : QTableWidget(parent)
+  setAlternatingRowColors(true);
 
   setFont(Settings::Instance().GetDebugFont());
+  QFontMetrics fm(Settings::Instance().GetDebugFont());
```
```suggestion
  const QFontMetrics fm(Settings::Instance().GetDebugFont());
```

lioncash commented on Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp:
```diff
@@ -72,7 +78,10 @@ void MemoryViewWidget::Update()
     setRowCount(1);
 
-  setRowHeight(0, 24);
+  QFontMetrics fm(Settings::Instance().GetDebugFont());
```
```suggestion
  const QFontMetrics fm(Settings::Instance().GetDebugFont());
```

TryTwo commented:
I noticed this doesn't work as well as code view's font based sizing, because it doesn't re-calibrate on font change/signal. I'll have to update it.

TryTwo commented:
Moved the verticalheader stuff into Update and tweaked it.
Run Update on font change. 
Added horizontalHeader()->setMinimumSectionSize to let BP column be smaller.

verticalHeader()->setDefaultSectionSize  appears to do what setRowHeight was doing just fine, so removed setRowHeight..
